src/main/java/com/joelj/jenkins/eztemplates/jobtypes/JobsFacade.java

```diff
@@ -0,0 +1,113 @@
+package com.joelj.jenkins.eztemplates.jobtypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.kohsuke.stapler.Ancestor;
+
+import com.google.common.base.Throwables;
+import com.joelj.jenkins.eztemplates.exclusion.Exclusions;
+
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.Label;
+import hudson.scm.SCM;
+import hudson.triggers.Trigger;
+import hudson.triggers.TriggerDescriptor;
+
+
+/**
+ * Facade to encapsulate the access to different types of Jenkins Jobs. Should be used when direct use of Job isn't possible and you have
+ * to deal with concrete implementations of Jobs.
+ *
+ * @see <a href="https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#historical-background">pipeline's DEVGUIDE.md#historical-background</a>
+ * @author Juan Pablo Santos Rodríguez
+ */
+public class JobsFacade {
+
+    private static boolean pipelinesPluginEnabled = Exclusions.checkPlugin("workflow-job") == null;
+
+    /**
+     * Verifies if the the plugin applies to the Jenkins job type.
+     *
+     * @param jobType Jenkins job type.
+     * @return {@code true} if it is either an {@link AbstractProject} or a {@link WorkflowJob}.
+     */
+    public static boolean isPluginApplicableTo( Class< ? extends Job > jobType ) {
+        if( pipelinesPluginEnabled ) {
+            return AbstractProject.class.isAssignableFrom( jobType )
+                || WorkflowJob.class.isAssignableFrom( jobType );
+        }
+        return AbstractProject.class.isAssignableFrom( jobType );
+    }
+
+    public static List< Job > getAllTemplatableJobs() {
+        List< Job > jobs = new ArrayList<>();
+        jobs.addAll( getApplicableJobOperationsFor( AbstractProject.class, null ).getAllJobs() );
+        jobs.addAll( getApplicableJobOperationsFor( WorkflowJob.class, null ).getAllJobs() );
+        return jobs;
+    }
+
+    public static List< Job > getAllJobs( Class< ? extends Job > jobType ) {
+        List< Job > jobs = new ArrayList<>();
+        jobs.addAll( getApplicableJobOperationsFor( jobType, null ).getAllJobs() );
+        return jobs;
+    }
+
+    public static Ancestor findTemplatableAncestorFrom(Ancestor ancestor) {
+        Ancestor job = ancestor;
+        while (job != null && ( !(job.getObject() instanceof AbstractProject) ||
+                                !(job.getObject() instanceof WorkflowJob) ) ) {
+            job = job.getPrev();
+        }
+        return job;
+    }
+
+    public static Label getAssignedLabel( Job job ) {

```

Comment:
Well, it was intentional. Either you use `Job`, or if the functionality is on one of the subclasses you go for a `JobProxy` (through the `JobsFacade`). This is intentional, in order to ease the incorporation of other types of Jobs, f.ex., if you would like to add support for `hudson.model.ViewJob`, one of the available Job types on Jenkins by default (which, from the javadoc, is a job that, `monitors activities that happen outside Hudson, which requires occasional batch reload activity to obtain the up-to-date information.`) you'd only have to implement `JobProxy`, and add support for it on one method on `JobsFacade`, and you'd be done, you don't have to touch any of the rest of the plugin. 

But my main point is about extension, going to `JobProxy` instances allows easier incorporation of new Job types. Also, things specific of Pipeline are encapsulted on their class, the same for AbstractProject, and the same for future types of Jobs. Other types of plugins which extend Job might find this handy. Just thought of how difficult would be to add support for another job type with the `JobProxy` implementations, or going through all the plugin. WDYT?

Another thought, about the no-ops: maybe have an abstract `JobProxy` base class with a generified `getAllJobs`, and the rest of the methods left with an empty implementation? That way the subclasses wouldn't contain no-ops.. (looking forward to the future, the day JENKINS-27299 gets fixed you'll be able to have 3 out of 9 methods as no-op on `PipelineProxy`, instead of the 5 as of today)